File: p1/part2/myfsm/fsm.pl
%dbase(fsm,[node,transition]).

%table(node,[nodeid,name,type]).
node(nStart, start, start).
node(nReady, ready, state).
node(nDrink, drink, state).
node(nEat, eat, state).
node(nFam, family, state).
node(nStop, stop, stop).

%table(transition,[transid,startsAt,endsAt]).
transition(t1, nStart, nReady).
transition(t2, nReady, nDrink).
transition(t3, nReady, nEat).
transition(t4, nDrink, nDrink).
transition(t5, nEat, nEat).
transition(t6, nDrink, nEat).
transition(t7, nEat, nDrink).
transition(t8, nDrink, nFam).
transition(t9, nEat, nFam).
transition(t10, nFam, nStop).

File: p1/part2/myfsm/fsm.java
package myfsm;

public class fsm {
   public enum states { start, ready, drink, eat, family, stop }

   states currentState = states.start;

   public String getName() { return currentState.toString(); }

   public void gotostart() {  
      switch(currentState) {
      default : 
         System.out.println("ignoring transition to start");
      }
   }
   public void gotoready() {  
      switch(currentState) {
      case start : 
         System.out.println("going to ready");
         currentState = states.ready; 
         break;
      default : 
         System.out.println("ignoring transition to ready");
      }
   }
   public void gotodrink() {  
      switch(currentState) {
      case ready : 
      case drink : 
      case eat : 
         System.out.println("going to drink");
         currentState = states.drink; 
         break;
      default : 
         System.out.println("ignoring transition to drink");
      }
   }
   public void gotoeat() {  
      switch(currentState) {
      case ready : 
      case drink : 
      case eat : 
         System.out.println("going to eat");
         currentState = states.eat; 
         break;
      default : 
         System.out.println("ignoring transition to eat");
      }
   }
   public void gotofamily() {  
      switch(currentState) {
      case drink : 
      case eat : 
         System.out.println("going to family");
         currentState = states.family; 
         break;
      default : 
         System.out.println("ignoring transition to family");
      }
   }
   public void gotostop() {  
      switch(currentState) {
      case family : 
         System.out.println("going to stop");
         currentState = states.stop; 
         break;
      default : 
         System.out.println("ignoring transition to stop");
      }
   }
}
